### ISSUE TITLE
Deprecate shipped 404 and 500 middlewares

### ISSUE THREAD

**Issue by jasisk** [krakenjs/kraken-js#359]
A couple of middleware ship in the `kraken-js` package but are—otherwise—not configured or enabled by default. They are able to be added to a project that uses `kraken-js` by being required:
```` js
var middlewareFactory = require('kraken-js/middleware.404');
app.use(middlewareFactory('templateName'));
```

These middleware **are** [presently configured by default](https://github.com/krakenjs/generator-kraken/blob/7553db517b7048d252afa593816dfe4192ce5c3a/app/templates/common/config/config.json#L61-L77) with the [yeoman generator](https://github.com/krakenjs/generator-kraken). However, it's been my experience that they cause more problems than they solve. Some of the problems we've seen involve assumptions based on xhr and not `content-type`, the assumption a rendering engine exists resulting in 404s turning into 500s, etc. Relying on the [default handler that ships with express](https://github.com/pillarjs/finalhandler) is sufficient while developing, while writing an appropriate handler for production ([404](http://expressjs.com/starter/faq.html#how-do-you-handle-404s), [5xx](http://expressjs.com/guide/error-handling.html)) should be standard practice.

**Comment by aredridel** [comment]
Burn them with fire.

**Comment by pvenkatakrishnan** [comment]
:+1:

**Comment by jasisk** [comment]
Just waiting on dougwilson/nodejs-depd#17.

**Comment by jasisk** [comment]
Closed by #361
